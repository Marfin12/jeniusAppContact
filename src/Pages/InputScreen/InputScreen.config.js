const defaultItem = {
    firstName: "",
    lastName: "",
    age: "",
    photo: "",
    id: ""
}

const defaultProps = {
    route: {
        params: {
            props: {
                item: defaultItem
            }
        }
    }
};
  
const displayName = 'InputScreen';
  
export default {
    defaultProps,
    defaultItem,
    displayName,
};