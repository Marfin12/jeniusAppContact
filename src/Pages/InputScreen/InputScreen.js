import * as React from 'react';
import { View } from 'react-native';
import { graphql, compose } from 'react-apollo';
import { get } from 'lodash';

import { PostContact, EditContact, GetContact } from '../../Graphql/contact.graphql';
import InputField from '../../Components/InputField';
import TextField from '../../Components/TextField';
import Card from '../../Components/Card';

import styles from './InputField.styles';

export const queryOptions = (props) => ({
  fetchPolicy: 'network-only',
  notifyOnNetworkStatusChange: true,
  variables: {
    id: get(props, 'navigation.state.params.props.data', ''),
  }
});

export const mapPostMutationToProps = ({ mutate }) => ({
  postContact: (firstName, lastName, age, photo) =>
    mutate({ variables: { input: { firstName, lastName, age, photo } } })
});

export const mapEditMutationToProps = ({ mutate }) => ({
  editContact: (firstName, lastName, age, photo) =>
    mutate({ variables: { input: { firstName, lastName, age, photo } } })
});

const _mutateContact = (contact, isEdit) => {
  const {firstName, lastName, age, photo} = contact;

  if (isEdit) editContact(firstName, lastName, age, photo);
  else postContact(firstName, lastName, age, photo);
};

const InputScreen = (props) => {
  const { isEdit, data } = props.navigation.state.params;
  const { firstName, lastName, age, photo } = data;

  const [firstNameText, setFirstName] = React.useState(firstName);
  const [lastNameText, setLastName] = React.useState(lastName);
  const [ageText, setAge] = React.useState(age);
  const [photoText, setPhoto] = React.useState(photo);
  const [contact, setContact] = React.useState({});

  React.useEffect(() => {
    setContact({
      firstName,
      lastName,
      age,
      photo
    })
  }, [firstName, lastName, age, photo])

  return (
    <View style={styles.container}>
      <InputField setText={setFirstName}>
        {firstNameText}
      </InputField>
      <InputField setText={setLastName}>
        {lastNameText}
      </InputField>
      <InputField setText={setAge}>
        {ageText}
       </InputField>
       <InputField setText={setPhoto}>
        {photoText}
       </InputField>
       <Card style={styles.button} onPress={_mutateContact(contact, isEdit)}>
        <TextField textStyle={styles.buttonTitle}>
          {isEdit ? 'Upadate' : 'Save'}
        </TextField>
      </Card>
    </View>
  );
};

export default compose(
  graphql(GetContact, {
    options: queryOptions
  }),
  graphql(PostContact, {
    props: mapPostMutationToProps
  }),
  graphql(EditContact, {
    props: mapEditMutationToProps
  })
)(InputScreen);
