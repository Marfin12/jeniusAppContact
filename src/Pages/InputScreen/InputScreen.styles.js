const styles = {
  container: {
    marginLeft: 10
  },
  button: {
    alignSelf: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 4,
    backgroundColor: 'black',
    flexDirection: 'column',
    flex: 0,
    shadowColor: '#ffd200',
    shadowOpacity: 0.8
  },
  buttonTitle: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
    color: 'white',
  },
  formInput: {
    marginTop: 70,
  },
  textTitle: {
    flex: 1,
    flexDirection: 'column',
    marginLeft: 12,
    paddingBottom: 100,
    justifyContent: 'center',
  },
};

export default styles;
