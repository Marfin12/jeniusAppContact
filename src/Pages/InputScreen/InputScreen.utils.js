import { Alert } from 'react-native';
import Constants from '../../Constants';

const _isSuccessResponse = (message) => message === "contact saved" || message === "Contact edited";

const _navigateToContactScreen = (isNavigation, navigation) => () => {
    if (isNavigation) {
        navigation.goBack();
    }
};

const _renderAlertDialog = (message, isNavigation, navigation) =>
  Alert.alert(
    _isSuccessResponse(message) ? "Success" : "Error",
    message,
    [
      { text: "OK", onPress: _navigateToContactScreen(isNavigation, navigation) }
    ]
  );

export const mutateContact = async (contact, isEdit, props) => {
    const {firstNameText, lastNameText, ageText, photoText, id, setIsLoading} = contact;
    setIsLoading(true);
    let message = "Oops!! something wrong!";
    let isNavigation = false;
    var result;
    try {
        if (isEdit) {
            result = await props.editContact(firstNameText, lastNameText, ageText, photoText, id);
            message = result.data.editContact.message;
        }
        else {
            result = await props.postContact(firstNameText, lastNameText, ageText, photoText);
            message = result.data.postContact.message;
        }
        setIsLoading(false);
        isNavigation = _isSuccessResponse(message);
    } catch(err) {
        setIsLoading(false);
        if (isEdit) {
            message = err.data.editContact.message;
        } else {
            message = err.data.postContact.message;
        }
    } finally {
        _renderAlertDialog(message, isNavigation, props.navigation);
    }
}
