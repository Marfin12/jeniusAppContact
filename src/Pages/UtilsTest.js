const apolloObjectCreator = (getListContact) => ({
    length: getListContact.length,
    getListContact,
  });
  
const apolloMapCreator = () => {
    const queryStore = new Map();
    queryStore.set({
      queryId: 123,
    });
  
    return queryStore;
};

export { apolloObjectCreator, apolloMapCreator };