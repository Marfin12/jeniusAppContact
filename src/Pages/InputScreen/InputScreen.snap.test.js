import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';

import InputScreen from './InputScreen';
import config from './InputScreen.config'
import { apolloMapCreator } from '../UtilsTest';

jest.mock('../../Components/TextField', () => 'TextField');
jest.mock('../../Components/Card', () => 'Card');
jest.mock('../../Components/FloatButton', () => 'FloatButton');
jest.mock('../../Components/Loading', () => 'Loading');
jest.mock('../../Components/InputField', () => 'InputField');

const navigation = {
  addListener: jest.fn()
};
const client = {
  watchQuery: jest.fn(),
  refetch: false,
};
const mockedApolloConfig = {
  refetch: jest.fn(),
  fetchMore: jest.fn(),
  updateQuery: jest.fn(),
  startPolling: jest.fn(),
  stopPolling: jest.fn(),
  subscribeToMore: jest.fn(),
  options: {
    fetchPolicy: 'no-cache',
  },
  queryManager: {
    queryStore: apolloMapCreator(),
  },
  queryId: 123,
  subscribe: jest.fn(),
  currentResult: {
    loading: false,
    partial: true,
    networkStatus: 'ready',
    errors: null,
    error: null,
  },
};

let InputComponent;

afterEach(() => {
  jest.clearAllMocks();
});

describe('Input Screen Component with loading', () => {
  jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    const lightMode = {
      theme: false,
    };
    const darkMode = {
      theme: true,
    };
    return {
      ...ActualReactRedux,
      useSelector: jest
        .fn()
        .mockImplementationOnce(() => lightMode)
        .mockImplementationOnce(() => darkMode),
    };
  });

  let mockedProps;

  beforeEach(() => {
    client.watchQuery = jest.fn(() => ({
      ...mockedApolloConfig,
      currentResult: jest.fn().mockImplementation(() => ({
        ...mockedApolloConfig.currentResult,
        loading: true,
      })),
    }));
    mockedProps = {
      route: {
        params: {
          isEdit: false,
          props: {
            item: config.defaultItem
          }
        }
      }
    }
  });

  test('renders light mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });

  test('renders dark mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });
});

describe('input Screen Component with failed to fetch', () => {
  jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    const lightMode = {
      theme: false,
    };
    const darkMode = {
      theme: true,
    };
    return {
      ...ActualReactRedux,
      useSelector: jest
        .fn()
        .mockImplementationOnce(() => lightMode)
        .mockImplementationOnce(() => darkMode),
    };
  });

  let mockedProps;

  beforeEach(() => {
    client.watchQuery = jest.fn(() => ({
      ...mockedApolloConfig,
      currentResult: jest.fn().mockImplementation(() => ({
        ...mockedApolloConfig.currentResult,
        loading: true,
      })),
    }));
    mockedProps = {
      route: {
        params: {
          isEdit: false,
          props: {
            item: config.defaultItem
          }
        }
      }
    }
  });

  test('renders light mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });

  test('renders dark mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });
});

describe('input Screen Component with successfully fetch', () => {
  jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    const lightMode = {
      theme: false,
    };
    const darkMode = {
      theme: true,
    };
    return {
      ...ActualReactRedux,
      useSelector: jest
        .fn()
        .mockImplementationOnce(() => lightMode)
        .mockImplementationOnce(() => darkMode),
    };
  });

  let mockedProps;

  beforeEach(() => {
    client.watchQuery = jest.fn(() => ({
      ...mockedApolloConfig,
      currentResult: jest.fn().mockImplementation(() => ({
        ...mockedApolloConfig.currentResult,
        loading: true,
      })),
    }));
    mockedProps = {
      route: {
        params: {
          isEdit: false,
          props: {
            item: config.defaultItem
          }
        }
      }
    }
  });

  test('renders light mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });

  test('renders dark mode input screen', () => {
    InputComponent = (
      <ApolloProvider client={client}>
        <InputScreen {...mockedProps} navigation={navigation} />
      </ApolloProvider>
    );

    const ScreenWrapper = renderer.create(InputComponent).toJSON();
    expect(ScreenWrapper).toMatchSnapshot();
  });
});
