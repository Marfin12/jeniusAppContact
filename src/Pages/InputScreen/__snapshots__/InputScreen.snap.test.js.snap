// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Component with failed to fetch renders dark mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;

exports[`Home Screen Component with failed to fetch renders light mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;

exports[`Home Screen Component with loading renders dark mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;

exports[`Home Screen Component with loading renders light mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;

exports[`Home Screen Component with successfully fetch renders dark mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;

exports[`Home Screen Component with successfully fetch renders light mode home screen 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Loading
    isVisible={false}
  />
  <View
    style={
      Object {
        "marginTop": 70,
      }
    }
  >
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      First Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Last Name:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Age:
    </InputField>
    <InputField
      setText={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 12,
          "paddingBottom": 100,
        }
      }
      text=""
    >
      Photo:
    </InputField>
    <Card
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 32,
          "paddingVertical": 12,
          "shadowColor": "#ffd200",
          "shadowOpacity": 0.8,
        }
      }
    >
      <TextField
        textStyle={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Save
      </TextField>
    </Card>
  </View>
</View>
`;
